RFCs/0011-partiql-insert-update-spec.md

```diff
@@ -0,0 +1,761 @@
+
+When attribute names are provided, they can be listed in any order. Each required attribute that is omitted in the provided attributes, MUST be filled with a default value, either its declared default value or an implementation of null. If the expression for any attribute is not of the correct data type, automatic type conversion i.e. type coercion MAY be attempted.
+
+#### 3.1 Examples
+
+In the following examples, `films` table has a closed schema.
+Insert a single row:
+
+```SQL
+INSERT INTO films
+VALUES ('UA502', 'Bananas', 105, '1971-07-13', 'Comedy', '82 minutes');
+```
+
+In this example, the `len` required attribute is omitted in the explicit list, and therefore it will have the default value:
+
+```SQL
+INSERT INTO films (code, title, did, date_prod, kind)
+VALUES ('T_601', 'Yojimbo', 106, '1961-06-16', 'Drama');
+```
+
+The following examples use the `DEFAULT` clause for the date attributes rather than specifying a value:
+
+```SQL
+INSERT INTO films
+VALUES ('UA502', 'Bananas', 105, DEFAULT, 'Comedy', '82 minutes');
+
+INSERT INTO films (code, title, did, date_prod, kind)
+VALUES ('T_601', 'Yojimbo', 106, DEFAULT, 'Drama');
+```
+
+The following examples statement inserts a row or item consisting entirely of default values for required attributes:
+
+```SQL
+INSERT INTO films DEFAULT VALUES;
+```
+
+The following example statement inserts multiple rows using the multi-row VALUES syntax:
+
+```SQL
+INSERT INTO films (code, title, did, date_prod, kind)
+VALUES ('B6717', 'Tampopo', 110, '1985-02-10', 'Comedy'),
+       ('HG120', 'The Dinner Game', 140, DEFAULT, 'Comedy');
+```
+
+In the following statement, we insert two values to `Music` table, `Music` table has an open schema with `Artist` and `SongTitle` as required attributes:
+
+```SQL
+INSERT INTO Music
+    <<{'Artist' : 'Acme Band', 'SongTitle' : 'PartiQL Rocks'},
+    {'Artist' : 'Emca Band', 'TitleSong' : 'PartiQL Rocks'}>>;
+```
+
+In the following statement we insert multiple person items as a bag value to `Person` table, `Person` table has an open schema with `LastName` and `DOB` as required attributes:
+
+```SQL
+INSERT INTO Person <<{'FirstName' : 'Raul',
+    'LastName' : 'Lewis',
+    'DOB' : 1963-08-19T,
+    'GovId' : 'LEWISR261LL',
+    'GovIdType' : 'Driver License',
+    },{'LastName' : 'Logan',
+    'DOB' : 1967-07-03T,
+    'Address' : '43 Stockert Hollow Road, Everett, WA, 98203'
+    },{'LastName' : 'Pena',
+    'DOB' : 1974-02-10T,
+    'GovId' : '744 849 301',
+    'GovIdType' : 'SSN',
+    'Address' : '4058 Melrose Street, Spokane Valley, WA, 99206'
+    }>>;
+```
+
+### 3.2 Query
+
+`SELECT` query expression can be used for inserting the resulting values from the query to the target table. The resulting tuples (E.g. rows) MUST have the same attribute name and type as the target schema’s required attributes.
```

Comment:
PartiQL/SQL++ should be generalizing this not to just be a `select` query but *any* expression returning any PartiQL value.

Reply:
I think it makes sense to extend the grammar incrementally but not necessarily for all PartiQL values. For example, not sure if scalars, or `MISSING` are applicable. Having said that, to your point, I think we need a follow-up issue for this; but at this stage—for consistency—I'd rather leaving it out of scope.